x/collection/keeper/grpc_query.go

```diff
@@ -85,6 +85,16 @@ func (s queryServer) AllBalances(c context.Context, req *collection.QueryAllBala
 
 		coin := collection.NewCoin(tokenID, balance)
 		balances = append(balances, coin)
+
+		if err := collection.ValidateNFTID(tokenID); err == nil {
+			s.keeper.iterateDescendants(ctx, req.ContractId, tokenID, func(tokenID string, _ int) (stop bool) {
+				coin := collection.NewCoin(tokenID, sdk.OneInt())
+				balances = append(balances, coin)
+
+				return false
+			})
+		}
```

Comment:
Why is this iteration necessary?

Reply:
Because daphne returns the whole nfts including the descendants. The DB of finschia only cares about the root nft for the performance and other reasons.

Reply:
It could be better to apply it on another PR.

Reply:
Removed the fix.